content/projects/gsoc/2020/projects/custom-jenkins-distribution-build-service.adoc

```diff
@@ -20,7 +20,52 @@ tags:
 - tools
 ---
 
-The main idea behind the project is to build a customizable jenkins distribution service that could be used to build tailor-made jenkins distributions. The service would provide users with a simple interface to select the configurations they want to build the instance with eg: plugins, authorization matrices etc. Furthermore it would include a section for sharing community created distros so that users can find and download already built jenkins war/configuration files to use out of the box.
+The main idea behind the project is to build a customizable Jenkins distribution service that could be used to build tailor-made Jenkins distributions. The service would provide users with a simple interface to select the configurations they want to build the instance with eg: plugins, authorization matrices etc. Furthermore it would include a section for sharing community created distros so that users can find and download already built Jenkins war/configuration files to use out of the box.
+
+=== Key Deliverables
+* A backend service built on spring boot having a modern Restful API support for constructing a  customized WAR package and configuration files.
+* A configuration constructing engine that can convert UI fields into a configuration file.
+* Front End application in the form of a website with a simple user interface for the users to interact with and customize the distribution.
+* Repository sharing community built most commonly used Jenkins configuration files.
+
+=== Plan for Phase 1 
+
+* Develop a simple workflow(Spring Boot Application) from UI bindings to the generation of the JCasC YAML file.
+  ** Reuse concepts from the JCasC Plugin to generate this JCasC file. 
+  ** Verify the configuration generated against the JCasC Schema.
+* Begin with a simple User Interface to test the workflow against.
+* Reuse logic from the jenkins core to construct and if possible render the user interface in react for all the plugins.
+* If the rendering is not possible, default to building the schemas from the data bound constructors or the JCasC Schema.
+* Blog Post for Phase 1.
+
+=== Plan for Phase 2
+
+* Utilize the custom war packager library for heavy lifting and converting the configuration file to a war package.
+* This project builds a wrapper around the CWP so some logic can be reused from here if necessary.
```

Comment:
```suggestion
* If necessary, implement a new wrapper around CWP so some logic can be reused from here
```

I do not think it will be needed, I plan to keep CWP in alpha for the time being so that changes can be integrated there directly if needed

Reply:
Yes